spec/requests/posts_spec.rb

```diff
@@ -0,0 +1,315 @@
+require "rails_helper"
+include SessionsHelper
+
+RSpec.describe PostsController, type: :controller do
+  let!(:user) { create(:user) }
+  let!(:user2) { create(:user) }
+  let!(:user3) { create(:user) }
+  let!(:other_user) { create(:user) }
+
+  let!(:posts) do
+    create_list(:post, 15, user: user, status: :published)
+  end
+
+  let!(:posts_other_user) do
+    create_list(:post, 5, user: other_user, status: :published)
+  end
+
+  let!(:limit) { Settings.limit.limit_10 }
+
+  shared_examples "when user is not owner" do
+    context "when user is not owner" do
```

Comment:
Khi nào dùng context, it và describe

Reply:
describe: để gom các test case lại với nhau, thường dùng để mô tả kiểm thử lớp hay phương thức, module...
context: để gom các test case, dùng để mô tả khi có điều kiện cụ thể hoặc là hoàn cảnh khác nhau mà một nhóm test case
it: mô tả 1 test case cụ thể của 1 hành động rồi cho ra kết quả mong muốn